### PR TITLE
issue #9408 fix (aws)

### PR DESCRIPTION
**Pull request by Albel-Analyst**
Should fix https://github.com/projectnessie/nessie/issues/9408 (aws only)

### CODE REVIEW

File: cli/cli/build.gradle.kts [hunk 1]
```diff
@@ -44,7 +44,8 @@ dependencies {
   // Include these FileIO implementations, as those are necessary to initialize
   // RESTCatalog *sigh*. Those FileIO's aren't functional, because the necessary
   // cloud specific clients (e.g. awssdk) are missing.
-  implementation("org.apache.iceberg:iceberg-aws")
+  implementation("org.apache.iceberg:iceberg-aws:1.6.1")
+  implementation("org.apache.iceberg:iceberg-aws-bundle:1.6.1")
```
Comment (dimas-b) [inline cli/cli/build.gradle.kts:48]: Please `gradle/libs.versions.toml` to manage library versions.

**Comment by dimas-b** [comment]
Thanks for you contribution @Albel-Analyst ! As discussed in Zulip, would you mind adding similar dependencies for GCS and ADLS to have a well-rounded fix?

Also, please put a short summary of changes in the PR title, e.g. something like `Add storage lib dependencies to CLI`. Thx!

File: cli/cli/build.gradle.kts (hunk 1 above)
Comment (adutra) [reply cli/cli/build.gradle.kts:48]: And while at it, I thinkw e can go ahead and add the three bundles:

```kotlin
  // Include these FileIO implementations, as those are necessary to initialize
  // RESTCatalog *sigh*.
  implementation("org.apache.iceberg:iceberg-aws")
  implementation("org.apache.iceberg:iceberg-aws-bundle")
  implementation("org.apache.iceberg:iceberg-azure")
  implementation("org.apache.iceberg:iceberg-azure-bundle")
  implementation("org.apache.iceberg:iceberg-gcp")
  implementation("org.apache.iceberg:iceberg-gcp-bundle")
```

Comment (adutra) [reply cli/cli/build.gradle.kts:48]: What we need to assess now is the jar size... it goes from 25Mb to 120Mb 🙄

Comment (snazy) [reply cli/cli/build.gradle.kts:48]: > What we need to assess now is the jar size... it goes from 25Mb to 120Mb 🙄

That's quite too big. I doubt a fix would need the whole Iceberg bundle, rather only the necessary dependencies.

**Comment by snazy** [comment]
Maybe 
```
  implementation("org.apache.iceberg:iceberg-aws")
  runtimeOnly(platform(libs.awssdk.bom))
  runtimeOnly("software.amazon.awssdk:s3")
  runtimeOnly("software.amazon.awssdk:sts")
  implementation("org.apache.iceberg:iceberg-gcp")
  runtimeOnly(platform(libs.google.cloud.storage.bom))
  runtimeOnly("com.google.cloud:google-cloud-storage")
  implementation("org.apache.iceberg:iceberg-azure")
  runtimeOnly(platform(libs.azuresdk.bom))
  runtimeOnly("com.azure:azure-storage-file-datalake")
```
?

**Comment by adutra** [comment]
There is a bunch of relocations in the bundles... will that work without them?

**Comment by dimas-b** [comment]
AFAIK relocations are critical only for Spark... It should be ok with relocations in Nessie CLI.

**Comment by snazy** [comment]
Should be good with #9640?

**Comment by Albel-Analyst** [comment]
I'm sorry, but I'm not a java programmer, so my corrections may be stupid, I see that the PR with corrections has already been filled in, so I'm closing this one, thank you

**Pull request closed without merging**